src/libecalc/core/models/compressor/train/simplified_train.py

```diff
@@ -517,12 +512,24 @@ def calculate_maximum_rate_given_outlet_pressure_single_calculation_point(
                     extrapolate_heads_below_minimum=False,
                 )
             )
-            polytropic_efficiency = float(
-                compressor_chart.efficiency_as_function_of_rate_and_head(
+
+            try:
+                efficiency_array: NDArray[np.float64] = compressor_chart.efficiency_as_function_of_rate_and_head(
                     rates=np.atleast_1d(maximum_actual_volume_rate),
                     heads=np.atleast_1d(polytropic_head),
                 )
-            )
+
+                if efficiency_array.size == 0:
+                    raise ValueError("Efficiency array is empty, cannot proceed with calculation.")
+
+                polytropic_efficiency = efficiency_array[0]
```

Comment:
should it therefore always be of length 1? Is it used for different cases, so in this case it is special and only an array of length 1 is expected?

Reply:
For all other compressor model `_evaluate_rate_ps_pd` has a for loop, so only floats are passed around from there. Maybe we should refactor simplified train to do the same...?

Reply:
I simplified error handling and I guess @jsolaas will give feedback on it.

Reply:
Not entirely sure what @jsolaas thought, but I also think that try...catch here is not needed as we do not add any more context to the exception, and we do not massage the exception to be sent directly to the user. As Jostein suggests, if we understand why this occurs here (exception), we can add that semantics and change to EcalcError instead where we now for sure that the exception has been handled, and it is safe to show e.g. the message to the user. The user should then probably understand what is the problem...

If we cannot say why the exception occurs, and we are unable to add more semantics/context to it, it is better to leave it (not try ... except)

Reply:
Ok. Then it will actually be like the way I did it the first time last week before I overwrote the commit and started adding unnecessary complexity. 

I will remove the try catch since there should be no way the array len >1 as long as polytropic_head is a float and the function efficiency_as_function_of_rate_and_head is working correctly.

Reply:
If @jsolaas and you others are ok with the latest change then its ready to be merged. Let me know it there is still some changes needed or if you see a need for the EcalcError to be used here.